modules/ding_nodelist/ding_nodelist.admin.inc

```diff
@@ -0,0 +1,186 @@
+    '#type' => 'fieldset',
+    '#collapsible' => FALSE,
+    '#title' => t('Autoscroll delay'),
+    '#description' => t('The delay to show an item for various nodelist widgets (in milliseconds).'),
+  );
+
+  $autoscrolling_widgets = array(
+    NL_WIDGET_CAROUSEL => array('Carousel', 3000),
+    NL_WIDGET_HOR_ACCORDION => array('Horizontal accordion', 3000),
+  );
+
+  foreach ($autoscrolling_widgets as $widget => $conf) {
+    $title = isset($conf[0]) ? check_plain($conf[0]) : '';
+    $form['ding_nodelist_autoscroll_delay']['ding_nodelist_' . $widget . '_delay'] = array(
+      '#type' => 'textfield',
+      '#title' => filter_xss($title),
+      '#default_value' => variable_get('ding_nodelist_' . $widget . '_delay', $conf[1]),
+      '#size' => 5,
+    );
+  }
+
+  return system_settings_form($form);
+}
+
+/**
+ * Template settings form.
+ */
+function ding_nodelist_templates_form($form, &$form_state) {
+
+  $form['templates'] = array(
+    '#type' => 'fieldset',
+    '#title' => t('Templates'),
+    '#tree' => TRUE,
+    '#theme' => '_ding_nodelist_templates_table',
+  );
+
+  $templates = _ding_nodelist_get_templates();
+
+  foreach ($templates as $template_name => $template) {
+    $form['templates'][$template_name]['template_name'] = array(
+      '#type' => 'hidden',
+      '#value' => $template_name,
+    );
+
+    $form['templates'][$template_name]['title'] = array(
+      '#type' => 'textfield',
+      '#default_value' => isset($template['title']) ? $template['title'] : '',
+      '#size' => 40,
+    );
+
+    $ct = node_type_get_names();
+    $form['templates'][$template_name]['content_type'] = array(
+      '#markup' => isset($ct[$template['content_type']]) ?
+      $ct[$template['content_type']] : t('Missing (@ctname)', array('@ctname' => $template['content_type'])),
+    );
+
+    $widgets = _ding_nodelist_get_widget_types();
+    $form['templates'][$template_name]['widget'] = array(
+      '#type' => 'select',
+      '#empty_option' => t('- select -'),
+      '#required' => FALSE,
+      '#default_value' => isset($template['widget']) ? $template['widget'] : '',
+      '#options' => $widgets,
+    );
+
+    // Visible in list of templates?
+    $form['templates'][$template_name]['visible'] = array(
+      '#type' => 'checkbox',
+      '#default_value' => $template['status'] == NL_TPL_ACTIVE ? TRUE : FALSE,
+    );
+  }
+
+  $form['submit'] = array(
+    '#type' => 'submit',
+    '#value' => t('Save'),
+  );
+
+  return $form;
+}
+
+/**
+ * Writes details about found templates.
+ */
+function ding_nodelist_templates_form_submit($form, &$form_state) {
+  foreach ($form_state['values']['templates'] as $template) {
+    $tpl_record = array(
+      'title' => $template['title'],
+      'status' => $template['visible'] ? NL_TPL_ACTIVE : NL_TPL_HIDDEN,
+      'widget' => $template['widget'],
+    );
+    db_update('ding_nodelist_templates')->fields($tpl_record)
+      ->condition('filename', $template['template_name'])->execute();
```

Comment:
Why store template configuration in a separate table? I had expected variables to be used like in for `ding_nodelist_trim_content_submit()`.